ansible-tests/validations/controller-ulimits-check.yaml

```diff
@@ -0,0 +1,34 @@
+---
+- hosts: controller
+  vars:
+    metadata:
+      name: Check controller ulimits
+      description: >
+        This will check the ulimits of each controller.
+      groups:
+      - post-deployment
+    nofiles_default: 1025
+    nproc_default: 1025
+  tasks:
+  - name: Get nofiles limit
+    become: true
+    command: sh -c "ulimit -n"
+    register: nofilesval
+    changed_when: False
+  - name: Check nofiles limit
+    fail:
+      msg: >
+        nofiles is set to {{ nofilesval.stdout }}.  It should be at least 2048 or higher, depending on available resources.
```

Comment:
I'm not sure I understand your comment. We're checking against the nofiles_min variable, the error message should be changed to use the nofiles_min variable too otherwise the test may fail with a misleading error if you have for example nofiles_min = 4096 and ulimit -n returns 2048.